src/Glpi/Form/Migration/FormMigration.php

```diff
@@ -106,6 +115,50 @@ public function getTypesConvertMap(): array
         ];
     }
 
+    public const PUBLIC_ACCESS_TYPE = 0;
+    public const PRIVATE_ACCESS_TYPE = 1;
+    public const RESTRICTED_ACCESS_TYPE = 2;
+
+    /**
+     * Get the class strategy to use based on the access type
+     *
+     * @return array Mapping between access type constants and strategy classes
+     */
+    public function getStrategyForAccessTypes(): array
+    {
+        return [
+            self::PUBLIC_ACCESS_TYPE => DirectAccess::class,
+            self::PRIVATE_ACCESS_TYPE => DirectAccess::class,
+            self::RESTRICTED_ACCESS_TYPE => AllowList::class
+        ];
+    }
+
+    /**
+     * Create the appropriate strategy configuration based on form access rights
+     *
+     * @param array $form_access_rights The access rights data from the database
+     * @return JsonFieldInterface The configuration object for the access control strategy
+     * @throws LogicException When no strategy config is found for the given access type
+     */
+    public function getStrategyConfigForAccessTypes(array $form_access_rights): JsonFieldInterface
+    {
+        $clean_ids = fn($ids) => array_unique(array_filter($ids, fn($id) => is_int($id)));
```

Comment:
I think using strict types in closure could PHPStan help to identifiy errors.

```suggestion
        $clean_ids = static fn(array $ids) => array_unique(array_filter($ids, fn(mixed $id) => is_int($id)));
```